content-claims.md

```diff
@@ -0,0 +1,334 @@
+
+When a publisher wants to hire a remote storage device this protocol design presents a challenge since these remote storage devices don’t speak native IPFS protocols, therefor IPFS clients can’t read content from them directly. We end up having to place a node between these remote storage devices and the client, proxying the data and incurring unnecessary egress.
+
+![](https://i.imgur.com/Jan0Opg.png)
+
+Content Claims change this.
+
+Publishers post **verifiable claims** about IPFS data they publish in remote storage devices into Content Discovery networks and services.
+
+From these claims, IPFS clients can request data directly from remote storage devices over standard transports like HTTP.
+
+![](https://i.imgur.com/4iutSwA.png)
+
+
+With content claims, any actor can provide data into the network without making the data available over an IPFS aware transport protocol. Rather than "serving" the data, content providers post verifiable claims about the data and its location.
+
+Clients can use these claims to read directly, over any existing transport (mostly HTTP), and in the act of reading the data will verify the related claims.
+
+This removes a **substantial** source of cost from providing content into the network. Content can be published into the network "at rest" on any permanent or temporary storage device that supports reading over HTTP.
+
+# Protocol
+
+## Claim Types
+
+The requirements for content claims break down into a few isolated components, each representing a specific claim. These claims are assembled together to represent the proof information necessary for retrieving content.
+
+All claim types map a **single** CID to "claim information."
+
+While you can derive block indexes from these claims (see: Inclusion Claims), each individual claim is indexed by a single **significant** CID (file root, dir root, etc) referrenced by `content` field.
+
+These claims include examples of a unixfs file  encoded into CAR files, but the protocol itself makes heavy use of CIDs in order to support a variety of future protocols and other use cases.
+
+Since this protocol builds upon the UCAN Invocation Specification, all of these claims are contained within a message from an *issuer* to a *destination*. This means it can be used to send specific actors unique route information apart from public content discovery networks, and can also be used to send messages to public content discovery networks by simply addressing them to a DID representing the discovery network.
+
+### Location Claims
+
+* Claims that a CID is available at a URL.
+* Block Level Interface, 
+  * GET Body MUST match multihash digest in CID.
+* Using CAR CID, or any future BlockSet() identifier, allows this to be a multi-block interface.
+
+```javascript
+{
+  "op": "assert/location",
+  "rsc": "https://web3.storage",
+  "input": {
+    "content" : CID /* CAR CID */, 
+    "location": "https://r2.cf/bag...car",
```

Comment:
note that implementation currently receives an array here, which I would think is also the best option :)